src/controllers/UserController.js

```diff
@@ -181,4 +182,55 @@ export default class UserController {
       return resError(res, 500, error.message);
     }
   }
+
+  /**
+ * @name forgetPassword
+ * @description User enters email address to receive a resetpassword link
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @returns {object} The API response
+ */
+  static async forgetPassword(req, res) {
+    try {
+      const { email } = req.body;
+      const { baseUrl } = req;
+      const { host } = req.headers;
+      const uri = host + baseUrl;
+      const { id } = await UserServices.getUserByEmail(email);
+      const token = Utils.generateToken({ email, id }, '20m');
```

Comment:
is there a way you check the '20m' you are sending if it has elapsed when the sends a new password with reset password link?

Reply:
I added a Middleware to verify the token and return 401 error code if the token is expired. Also in the reset password controller the token is verified, so if it's expired it will return an error message. But the Middleware takes care of that so it doesn't get to the controller before that's verified

Reply:
ok